copy/multiple.go

```diff
@@ -27,23 +32,90 @@ const (
 type instanceCopy struct {
 	op           instanceCopyKind
 	sourceDigest digest.Digest
+
+	// Fields which can be used by callers when operation
+	// is `instanceCopyClone`
+	compressionVariant            *compression.Algorithm
+	cloneAddPlatform              *imgspecv1.Platform
+	cloneAddAnnotations           map[string]string
+	cloneAddCompressionAlgorithms []compression.Algorithm
+}
+
+// internal type only to make imgspecv1.Platform comparable
+type platformComparable struct {
+	Architecture string
+	OS           string
+	OSVersion    string
+	OSFeatures   string
+	Variant      string
+}
+
+// Converts imgspecv1.Platform to a comparable format.
+func platformV1ToPlatformComparable(platform imgspecv1.Platform) platformComparable {
+	return platformComparable{Architecture: platform.Architecture,
+		OS:         platform.OS,
+		OSFeatures: strings.Join(platform.OSFeatures, ","),
+		OSVersion:  platform.OSVersion,
+		Variant:    platform.Variant,
+	}
+}
+
+// platformCompresionPair prepares a mapping of platformComparable -> CompressionAlgorithmNames for given digests
+func platformCompresionPair(list internalManifest.List, instanceDigests []digest.Digest) (map[platformComparable]*set.Set[string], error) {
+	platformCompressionMap := make(map[platformComparable]*set.Set[string])
+	for _, instanceDigest := range instanceDigests {
+		instanceDetails, err := list.Instance(instanceDigest)
+		if err != nil {
+			return nil, fmt.Errorf("getting details for instance %s: %w", instanceDigest, err)
+		}
+		platform := platformV1ToPlatformComparable(*instanceDetails.ReadOnly.Platform)
+		_, valueExists := platformCompressionMap[platform]
+		if !valueExists {
+			platformCompressionMap[platform] = set.New[string]()
+		}
+		set := platformCompressionMap[platform]
```

Comment:
Non-blocking:
```
platformSet, ok := …Map[platform]
if !ok {
   platformSet = set.New…
   …Map[platform] = platformSet
}
```
would avoid one map lookup, in both paths (OTOH this is clearly not going to be performance-critical), and it would make it clear that the `set := ` lookup here can never fail and return `nil`.